# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.h
# Commit: e9fc94b33da9
# Full Hash: e9fc94b33da9dc047b37e8bd2efe7f6b20fda079
# Author: Peter Van der Beken <peterv@propagandism.org>
# Date: 2021-12-14 21:34:09
# Regressor Bug: 1742865
# File Overlap Count: 1
# Description:
#   Bug 1742865 - Handle meta refresh correctly with session history in the parent. r=smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D132837
# ==============================================================================

diff -r f5bcc7acaf26 -r e9fc94b33da9 docshell/base/BrowsingContext.h
--- a/docshell/base/BrowsingContext.h	Tue Dec 14 10:10:32 2021 +0000
+++ b/docshell/base/BrowsingContext.h	Tue Dec 14 10:13:58 2021 +0000
@@ -791,10 +791,13 @@
   // context or any of its ancestors.
   bool IsPopupAllowed();
 
+  // aCurrentURI is only required to be non-null if the load type contains the
+  // nsIWebNavigation::LOAD_FLAGS_IS_REFRESH flag and aInfo is for a refresh to
+  // the current URI.
   void SessionHistoryCommit(const LoadingSessionHistoryInfo& aInfo,
-                            uint32_t aLoadType, bool aHadActiveEntry,
-                            bool aPersist, bool aCloneEntryChildren,
-                            bool aChannelExpired);
+                            uint32_t aLoadType, nsIURI* aCurrentURI,
+                            bool aHadActiveEntry, bool aPersist,
+                            bool aCloneEntryChildren, bool aChannelExpired);
 
   // Set a new active entry on this browsing context. This is used for
   // implementing history.pushState/replaceState and same document navigations.
@@ -898,6 +901,9 @@
     return mChildSessionHistory.forget();
   }
 
+  static bool ShouldAddEntryForRefresh(nsIURI* aCurrentURI,
+                                       const SessionHistoryInfo& aInfo);
+
  private:
   void Attach(bool aFromIPC, ContentParent* aOriginProcess);
 